src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleGroupSyncMessageReceiver.scala

```diff
@@ -0,0 +1,88 @@
+package org.broadinstitute.dsde.workbench.sam.google
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.effect.unsafe.implicits.global
+import com.google.cloud.pubsub.v1.{AckReplyConsumer, MessageReceiver}
+import com.google.common.annotations.VisibleForTesting
+import com.google.pubsub.v1.PubsubMessage
+import com.typesafe.scalalogging.LazyLogging
+import io.opencensus.trace.AttributeValue
+import net.logstash.logback.argument.StructuredArguments
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.model.FullyQualifiedPolicyId
+import org.broadinstitute.dsde.workbench.sam.util.OpenCensusIOUtils._
+import org.broadinstitute.dsde.workbench.sam.util.SamRequestContext
+import spray.json._
+
+import scala.util.Try
+
+/** Created by dvoet on 12/6/16.
+  */
+class GoogleGroupSyncMessageReceiver(groupSynchronizer: GoogleGroupSynchronizer) extends MessageReceiver with LazyLogging {
+  override def receiveMessage(message: PubsubMessage, consumer: AckReplyConsumer): Unit =
+    traceIO("GoogleGroupSyncMessageReceiver-PubSubMessage", SamRequestContext()) { samRequestContext =>
+      val groupId: WorkbenchGroupIdentity = parseMessage(message)
+      logger.info(s"received sync message: $groupId")
+      samRequestContext.parentSpan.foreach(
+        _.putAttribute("groupId", AttributeValue.stringAttributeValue(groupId.toString))
+      )
+      groupSynchronizer
+        .synchronizeGroupMembers(
+          groupId,
+          Set.empty,
+          samRequestContext
+        )
+        .redeem(syncFailure(_, consumer), syncComplete(_, consumer))
+    }.unsafeRunSync()
```

Comment:
Same here, having unsafeRunSync in the receive message could end up with dead letters